src/librustc_typeck/diagnostics.rs

```diff
@@ -4705,13 +4689,10 @@ register_diagnostics! {
 //  E0372, // coherence not object safe
     E0377, // the trait `CoerceUnsized` may only be implemented for a coercion
            // between structures with the same definition
-    E0521, // redundant auto implementations of trait
     E0533, // `{}` does not name a unit variant, unit struct or a constant
 //  E0563, // cannot determine a type for this `impl Trait`: {} // removed in 6383de15
     E0564, // only named lifetimes are allowed in `impl Trait`,
            // but `{}` was found in the type `{}`
-    E0567, // auto traits can not have type parameters
-    E0568, // auto-traits can not have predicates,
```

Comment:
Could you comment out the obsoleted error codes instead of removing them?

Reply:
Tidy is merciless, it gives me duplicate error codes even if it's commented out.

Reply:
Ah, right, they are not removed, just moved to AST validation.